xds/src/main/java/io/grpc/xds/CdsLoadBalancer.java

```diff
@@ -0,0 +1,290 @@
+      return true;
+    }
+
+    @Override
+    public int getPriority() {
+      return 5;
+    }
+
+    // A synthetic policy name identified by CDS config.
+    @Override
+    public String getPolicyName() {
+      return "cds_policy__cluster_name_" + cdsConfig.name;
+    }
+
+    @Override
+    public LoadBalancer newLoadBalancer(final Helper helper) {
+      return new LoadBalancer() {
+        // Becomes non-null once handleResolvedAddresses() successfully.
+        // Assigned at most once.
+        @Nullable
+        ClusterWatcherImpl clusterWatcher;
+
+        @Override
+        public void handleNameResolutionError(Status error) {}
+
+        @Override
+        public boolean canHandleEmptyAddressListFromNameResolution() {
+          return true;
+        }
+
+        @Override
+        public void shutdown() {
+          if (clusterWatcher != null && clusterWatcher.edsBalancer != null) {
+            clusterWatcher.edsBalancer.shutdown();
+          }
+        }
+
+        @Override
+        public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+          if (clusterWatcher == null) {
+            clusterWatcher = new ClusterWatcherImpl(helper, resolvedAddresses);
+            xdsClient.watchClusterData(cdsConfig.name, clusterWatcher);
+            if (oldCdsConfig != null) {
+              xdsClient.cancelClusterDataWatch(oldCdsConfig.name, oldClusterWatcher);
+            }
+            CdsLoadBalancer.this.clusterWatcher = clusterWatcher;
+          }
+          // Else handleResolvedAddresses() already called, so no-op here because the config is
+          // fixed in this balancer.
+        }
+      };
+    }
+  }
+
+  private final class ClusterWatcherImpl implements ClusterWatcher {
+
+    final Helper helper;
+    final ResolvedAddresses resolvedAddresses;
+
+    // EDS balancer for the cluster.
+    // Becomes non-null once handleResolvedAddresses() successfully.
+    // Assigned at most once.
+    @Nullable
+    LoadBalancer edsBalancer;
+
+    // LoadStatsStore for the cluster.
+    // Becomes non-null once handleResolvedAddresses() successfully.
+    // Assigned at most once.
+    @Nullable
+    LoadStatsStore loadStatsStore;
+
+    ClusterWatcherImpl(Helper helper, ResolvedAddresses resolvedAddresses) {
+      this.helper = helper;
+      this.resolvedAddresses = resolvedAddresses;
+    }
+
+    @Override
+    public void onClusterChanged(ClusterUpdate newUpdate) {
+      checkArgument(
+          newUpdate.getLbPolicy().equals("round_robin"),
+          "The load balancing policy in ClusterUpdate '%s' is not supported", newUpdate);
+
+      final XdsConfig edsConfig = new XdsConfig(
+          /* balancerName = */ null,
+          new LbConfig(newUpdate.getLbPolicy(), ImmutableMap.<String, Object>of()),
+          /* fallbackPolicy = */ null,
+          /* edsServiceName = */ newUpdate.getEdsServiceName(),
+          /* lrsServerName = */ newUpdate.getLrsServerName());
```

Comment:
Do we need the commented param names when it should be obvious from the expression value? For lines 230-231